maps/_zombiemode.gsc

```diff
@@ -4176,9 +4194,30 @@ round_think()
 	level.dog_health = 1600;
 	level.dog_round_count = 5;
 	level.game_started = 1;
-	level.next_dog_round = 666;
-	level.next_monkey_round = 666;
-	level.next_doc_round = 666;
+
+	// lveez - if don't wait for this flag the next doc rounds gets reset
+	if (level.script == "zombie_pentagon")
```

Comment:
This section for handling special rounds at the being at the beginning of `round_think()` means that it is only going to get executed once when the game is setting things up for the first time when loading into a map or restarting level. I think this would do better a couple lines down inside of the infinite loop before the flag for "round_start". That way, people can choose on the fly how often they want the next special round to occur (e.g. I only want to practice monkey rounds on Ascension, I can do them back-to-back without having to wait the typical 4-5 rounds to practice them again). The only thing you'll need to look out for is that the original logic for the next special round does not overwrite the custom logic. Otherwise everything else is good.

Reply:
In the latest commit I changed the logic so that you can change the frequency of the dog rounds at any point, and gave options of: 4/5 rounders, 4 rounders, 5 rounders, every round, and off. I think this would cover all use cases.

I also fixed the bug properly since the old commit didn't actually fix it despite it working every test I did originally.